mvp_onboarding.Server/Classes/CustomerMethods.cs

```diff
@@ -0,0 +1,118 @@
+﻿using mvp_onboarding.Server.Dtos;
+using mvp_onboarding.Server.Interfaces;
+using mvp_onboarding.Server.Models;
+using Microsoft.EntityFrameworkCore;
+using mvp_onboarding.Server.Mappers;
+
+namespace mvp_onboarding.Server.Classes
+{
+    public class CustomerMethods : ICustomerMethods
+    {
+        public CustomerMethods(TalentOnboardingContext context) 
+        {
+            _context = context;
+        }
+
+        private readonly TalentOnboardingContext _context;
+
+        public async Task<CustomerResponseDto> GetCustomers(int pageNumber, int pageSize, string sortColumn, string sortDirection)
+        {
+            var query = _context.Customers.AsQueryable();
+
+            try
+            {
+                if (sortDirection.ToLower() == "asc")
+                {
+                    query = query.OrderBy(c => EF.Property<Customer>(c, sortColumn));
+                }
+                else
+                {
+                    query = query.OrderByDescending(c => EF.Property<Customer>(c, sortColumn));
+                }
+            }
+            catch (InvalidOperationException ex)
+            {
+                Console.WriteLine(ex.Message);
+            }
+
+            var totalCount = await query.CountAsync();
+            var customers = await query
+                .Skip((pageNumber -1) * pageSize)
+                .Take(pageSize)
+                .ToListAsync();
+
+            var customerDtos = customers.Select(c => CustomerMapper.EntityToDto(c)).ToList();
+
+            return new CustomerResponseDto
+            {
+                Items = customerDtos,
+                TotalCount = totalCount,
+                PageSize = pageSize,
+                CurrentPage = pageNumber
+            };
```

Comment:
You can actually add this inside the **try/catch**. This might cause an error on your application if there is something wrong on this part.